### DISCUSSION TITLE
Not able to retrieve records from ES for generated DIAL codes

### DISCUSSION THREAD

**Discussion by Debayanpaul1593** [Sunbird-Knowlg/Community#110] [General Discussions]
Hi Sunbird Team,
I am facing a problem where I am unable to search for DIAL codes. The API and its details can be found below

**API endpoint:** <dial_service_base_url>/dialcode/v3/search
**Request Body:** 
`
{
    "request": {
        "search": {
        "identifier": "J9S4M5",
            "status": "Draft",
            "limit":5
        },
        "fields": ["dialcode_index","identifier","channel","batchcode"],
        "limit": 5,
        "sort": {"dialcode_index": "Desc"}
    }
}
`
**Response**
`
{
    "id": "sunbird.dialcode.search",
    "ver": "3.0",
    "ts": "2023-03-20T17:55:09Z+05:30",
    "params": {
        "resmsgid": "faa2afc9-eda3-4f4b-87de-28347774e8f7",
        "msgid": null,
        "err": "SYSTEM_ERROR",
        "status": "failed",
        "errmsg": "Elasticsearch exception [type=search_phase_execution_exception, reason=all shards failed]"
    },
    "responseCode": "SERVER_ERROR",
    "result": {}
}
`
After looking at a few solutions suggested on StackOverflow I was unable to come to a conclusion on how to solve this issue. 
I can see the shard status as green for all the indices as

green open compositesearch ajM9zRXXS_ad9g1yruDbig 1 0 0 0 261b 261b
green open dialcodemetrics 9Ef6lmMsSbeAO7CEijbAmg 1 0 0 0 261b 261b
green open dialcode        5ePYl53aTUO1E3xHviMdug 1 0 0 0 261b 261b

Can someone please provide some insight as to what could be the issue in this case? I am running Elastic Search 6.8.22 through Docker

**Comment by pallakartheekreddy** [comment]
@Debayanpaul1593 looks like some issue with `Elasticsearch`. Can you please share Elasticsearch logs to analyse the issue.
